ArjunaCore/arjuna/classes/com/arjuna/ats/arjuna/coordinator/BasicAction.java

```diff
@@ -1558,7 +1560,10 @@ protected synchronized int End (boolean reportHeuristics)
             {
                 if (heuristicDecision != TwoPhaseOutcome.HEURISTIC_ROLLBACK)
                 {
-                    TxStats.getInstance().incrementCommittedTransactions();
+                    if (startTime == null)
```

Comment:
stats can be enabled dynamically so they they can be disabled when the txn is started and then enabled mid flight.